add_on_usage_test.go

```diff
@@ -0,0 +1,267 @@
+		}
+	})
+
+	// Ensure a 404 returns nil values.
+	t.Run("ErrNotFound", func(t *testing.T) {
+		client, s := recurly.NewTestServer()
+		defer s.Close()
+
+		s.HandleFunc("GET", "/v2/subscriptions/1122334455/add_ons/addOnCode/usage/8888", func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusNotFound)
+		}, t)
+
+		if a, err := client.AddOnUsages.Get(context.Background(), "1122334455", "addOnCode", "8888"); !s.Invoked {
+			t.Fatal("expected fn invocation")
+		} else if err != nil {
+			t.Fatal(err)
+		} else if a != nil {
+			t.Fatalf("expected nil: %#v", a)
+		}
+	})
+}
+
+func TestAddOnUsage_Create(t *testing.T) {
+	client, s := recurly.NewTestServer()
+	defer s.Close()
+
+	s.HandleFunc("POST", "/v2/subscriptions/1122334455/add_ons/addOnCode/usage", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusCreated)
+		w.Write(MustOpenFile("add_on_usage.xml"))
+	}, t)
+
+	if a, err := client.AddOnUsages.Create(context.Background(), "1122334455", "addOnCode", recurly.AddOnUsage{}); !s.Invoked {
+		t.Fatal("expected fn invocation")
+	} else if err != nil {
+		t.Fatal(err)
+	} else if diff := cmp.Diff(a, NewTestAddOnUsage()); diff != "" {
+		t.Fatal(diff)
+	}
+}
+
+func TestAddOnUsage_Update(t *testing.T) {
+	client, s := recurly.NewTestServer()
+	defer s.Close()
+
+	s.HandleFunc("PUT", "/v2/subscriptions/1122334455/add_ons/addOnCode/usage/1234", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		w.Write(MustOpenFile("add_on_usage.xml"))
+	}, t)
+
+	if a, err := client.AddOnUsages.Update(context.Background(), "1122334455", "addOnCode", "1234", recurly.AddOnUsage{}); !s.Invoked {
+		t.Fatal("expected fn invocation")
+	} else if err != nil {
+		t.Fatal(err)
+	} else if diff := cmp.Diff(a, NewTestAddOnUsage()); diff != "" {
+		t.Fatal(diff)
+	}
+}
+
+func TestAddOnUsage_Delete(t *testing.T) {
+	client, s := recurly.NewTestServer()
+	defer s.Close()
+
+	s.HandleFunc("DELETE", "/v2/subscriptions/1122334455/add_ons/addOnCode/usage/1234", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusNoContent)
+	}, t)
+
+	if err := client.AddOnUsages.Delete(context.Background(), "1122334455", "addOnCode", "1234"); !s.Invoked {
+		t.Fatal("expected fn invocation")
+	} else if err != nil {
+		t.Fatal(err)
+	}
+}
+
+
+// Returns add on corresponding to testdata/add_on.xml
+func NewTestAddOnUsage() *recurly.AddOnUsage {
+
+	now := time.Date(2000, 1, 1, 0, 0, 0, 0, time.UTC)
+	return &recurly.AddOnUsage{
+		XMLName:            xml.Name{Local: "usage"},
+		Amount:             1,
+		MerchantTag:        "Order ID: 4939853977878713",
+		RecordingTimestamp: recurly.NewTime(now),
+		UsageTimestamp:     recurly.NewTime(now),
+		CreatedAt:          recurly.NewTime(now),
+		UpdatedAt:          recurly.NullTime{},
+		BilledAt:           recurly.NullTime{},
+		UsageType:          "price",
+		UnitAmountInCents:  45,
+		UsagePercentage:    recurly.NullFloat{},
```

Comment:
I'd recommend setting a value here (and in the XML file) since this is the only place in the repo where `NullFloat` is used within another struct.

Reply:
Good idea, thanks - changed